src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SqlInternalConnectionTds.cs

```diff
@@ -1118,7 +1118,7 @@ bool isDelegateControlRequest
                     }
                     else if (internalTransaction.OpenResultsCount != 0)
                     {
-                        //throw SQL.CannotCompleteDelegatedTransactionWithOpenResults(this);
+                        throw SQL.CannotCompleteDelegatedTransactionWithOpenResults(this, _parser.MARSOn);
```

Comment:
How does this change impact if MARS is enabled? Can you try to reproduce issue with MARS on with this fix?